source/common/buffer/buffer_impl.h

```diff
@@ -100,7 +100,7 @@ class Slice {
     if (available_size == 0) {
       return {nullptr, 0};
     }
-    uint64_t reservation_size = std::min(size, available_size);
+    size_t reservation_size = std::min(size, available_size);
```

Comment:
Can you cast this where it's actually required? I don't think it should be here?